src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -550,7 +550,11 @@ private static async Task<RestoreSummary> PerformNuGetV2RestoreAsync(Common.ILog
             {
                 cacheContext.NoCache = noCache;
 
-                var downloadContext = new PackageDownloadContext(cacheContext, repositoryPath, directDownload: false)
+                PackageNamespacesConfiguration packageNamespacesConfiguration = settings != null ?
```

Comment:
Can settings be null?

Reply:
In above codepaths it's used as never null.

Reply:
You're right, addressing now.